### PR TITLE
Add same_as validator

### PR DESCRIPTION
**Pull request by glaucocustodio**
Implements https://github.com/ruby-grape/grape/issues/1846.

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -20,6 +20,7 @@
   config.include Rack::Test::Methods
   config.include Spec::Support::Helpers
   config.raise_errors_for_deprecations!
+  config.filter_run_when_matching :focus
```
Comment (glaucocustodio) [inline spec/spec_helper.rb]: This line is out of the scope from this PR,  but I find it to be very useful. Can we keep?

**Comment by dblock** [comment]
Great stuff, merged. You can take this from here and make `greater_than`, `less_than`, etc.?

**Comment by dblock** [comment]
No, because I mean it should be relative to another value, eg.

```ruby
params do
   requires :first
   requires :last, greater_than: :first
end
```

**Comment by dblock** [comment]
Related, it might make sense to extract some of these validators into separate gems at some point. We could have a whole bunch of mathematical validators, password strength validators, etc.

**Comment by glaucocustodio** [comment]
Ok, I thought you were talking about length.. Gt/Lt make sense in this case.

I agree with the idea of extracting validators in another gem in the future..

**Pull request merged**